src/log.rs

```diff
@@ -0,0 +1,38 @@
+use std::convert::TryInto;
+
+use libc::c_void;
+
+use crate::{log_callback_get, rslice::rustls_str};
+
+struct Logger {}
+
+impl log::Log for Logger {
+    fn enabled(&self, _metadata: &log::Metadata<'_>) -> bool {
+        true
+    }
+    fn log(&self, record: &log::Record<'_>) {
+        if let Ok((Some(cb), userdata)) = log_callback_get() {
+            let message = format!("{} {} {}", record.target(), record.level(), record.args());
```

Comment:
Passing level separately as a defined int/enum is definitely useful to the application.